examples/apps/image-folder-support/app.py

```diff
@@ -0,0 +1,256 @@
+import logging
+import tempfile
+from dataclasses import dataclass
+from functools import cached_property
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import PIL.Image
+import plotly.express as px
+import streamlit as st
+from PIL.Image import Image
+from streamlit_image_select import image_select
+
+from landingai.common import Prediction, SegmentationPrediction
+from landingai.postprocess import (
+    class_counts,
+    class_map,
+    segmentation_class_pixel_coverage,
+)
+from landingai.predict import Predictor
+from landingai.visualize import overlay_predictions
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s %(filename)s %(funcName)s %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+_SUPPORTED_IMAGE_FORMATS = ["jpg", "jpeg", "png", "tiff"]
+
+
+@dataclass(frozen=True)
+class ImageInferenceResult:
+    local_image_path: Path
+    image: np.ndarray
+    predictions: list[Prediction]
+    img_with_preds_path: Path
+
+    @cached_property
+    def class_counts(self) -> dict[int, tuple[int, str]]:
+        return class_counts(self.predictions)
+
+    @cached_property
+    def image_with_predictions(self) -> Image:
+        return overlay_predictions(self.predictions, self.image)
+
+
+@dataclass(frozen=True)
+class ImageFolderResult:
+    image_folder: Path
+    image_predictions: list[ImageInferenceResult]
+
+    @cached_property
+    def class_map(self) -> dict[int, str]:
+        all_predictions = []
+        for res in self.image_predictions:
+            all_predictions.extend(res.predictions)
+        return class_map(all_predictions)
+
+    @cached_property
+    def dataframe(self) -> pd.DataFrame:
+        predictions = self.image_predictions
+        images = [pred.image for pred in predictions]
+        image_paths = [str(pred.local_image_path) for pred in predictions]
+        class_counts_all = {
+            class_name: [0] * len(images) for class_name in self.class_map.values()
+        }
+        for i, res in enumerate(predictions):
+            for count, class_name in res.class_counts.values():
+                class_counts_all[class_name][i] = count
+
+        data = {
+            "image_path": image_paths,
+        }
+        data.update(class_counts_all)
+        return pd.DataFrame(data)
+
+
+def bulk_inference(image_paths: list[str], image_folder_path: str) -> ImageFolderResult:
```

Comment:
I think this function would be ideal for the pipeline construct... in fact you are doing a very similar thing to accumulate results on a FrameSet

Reply:
good point, just add asana task to simplify and refactor this part.
https://app.asana.com/0/0/1204761250364986/f